src/App.js

```diff
@@ -1,21 +1,25 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
+import {BrowserRouter, Switch, Route} from "react-router-dom";
+import ContainerListGiphy from './giphy/container/container-list-giphy';
+import ContainerGiphysFavorites from './giphy-favorite/container/container-giphys-favorites';
+import NotFound from './giphy/components/not-found';
 import './App.css';
 
-class App extends Component {
+
+export default class App extends Component {
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
+      <React.Fragment>
+        <BrowserRouter>
+          <Switch>
+            <Route exact path="/" component={ ContainerListGiphy } />
+            <Route path="/favorites" component={ ContainerGiphysFavorites } />
+            <Route path='*' component={NotFound} />
```

Comment:
Not necessary to specify a path, if you live the path empty it will always match, and because the Route is at the end it will work as expected.